examples/common/golioth_basics.c

```diff
@@ -46,9 +46,9 @@ static golioth_rpc_status_t on_multiply(
     if (cJSON_GetArraySize(params) != 2) {
         return RPC_INVALID_ARGUMENT;
     }
-    int a = cJSON_GetArrayItem(params, 0)->valueint;
-    int b = cJSON_GetArrayItem(params, 1)->valueint;
-    snprintf((char*)detail, detail_size, "{ \"value\": %d }", a * b);
+    double a = cJSON_GetArrayItem(params, 0)->valuedouble;
+    double b = cJSON_GetArrayItem(params, 1)->valuedouble;
+    snprintf((char*)detail, detail_size, "{ \"value\": %lf }", a * b);
```

Comment:
In some configurations `snprintf()` does not handle printing floats. This is true for example with minimal picolibc configurations. In that case following string was generated: `{ "value": *float* }`, which is invalid and is not gracefully handled by cloud (obviously).

So I wonder if we should abandon this change instead and rely on integers.

@ncmiller thoughts?

Reply:
Yeah, I think it probably makes sense to abandon this change, so we can be compatible with minimal libc implementations like `picolibc`.